src/MSBuild/XMake.cs

```diff
@@ -2545,6 +2546,42 @@ private static bool WarningsAsErrorsSwitchIsEmpty(CommandLineSwitches commandLin
             return indexOfColon < 0 || indexOfColon == val.Length - 1;
         }
 
+        internal static ProjectIsolationMode ProcessIsolateProjectsSwitch(string[] parameters)
+        {
+
+            // Before /isolate had parameters, it was treated as a boolean switch.
+            // Preserve that in case anyone is using /isolate:{false|true}
+            if (parameters.Length == 1 && bool.TryParse(parameters[0], out bool boolValue))
+            {
+                return boolValue ? ProjectIsolationMode.True : ProjectIsolationMode.False;
+            }
+
+            ProjectIsolationMode isolateProjects = ProjectIsolationMode.True;
+            foreach (string parameter in parameters)
+            {
+                if (string.IsNullOrWhiteSpace(parameter))
+                {
+                    continue;
+                }
+
+                var messageUponIsolationViolationParams = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+                {
+                    ProjectIsolationMode.MessageUponIsolationViolation.ToString(),
+                    "Message",
+                };
+                if (messageUponIsolationViolationParams.Contains(parameter.Trim()))
+                {
```

Comment:
I think I'd rather see two `if` clauses than this allocation.
```suggestion
                string trimmedParameter = parameter.Trim();
                
                if (trimmedParameter.Equals(nameof(ProjectIsolationMode.MessageUponIsolationViolation), StringComparison.OrdinalIgnoreCase) ||
                    trimmedParameter.Equals("Message", StringComparison.OrdinalIgnoreCase))
                {
```